batch-nft-reveal/app/lib/deploy.ts

```diff
@@ -0,0 +1,33 @@
+import { ContractFactory, Signer } from 'ethers'
+
+import NFTCollection from '../artifacts/contracts/NFTCollection.sol/NFTCollection.json'
+import {
+  VRF_CALLBACK_GAS_LIMIT,
+  VRF_COORDINATOR_V2_ADDRESS,
+  VRF_GAS_LANE,
+} from '../conf/config'
+import NFTCollectionParams from './types/NFTCollectionParams'
+
+export async function deployNFTCollection(
+  nftParams: NFTCollectionParams,
+  signer: Signer,
+  chainId: number
+) {
+  const nftCollection = new ContractFactory(
+    NFTCollection.abi,
+    NFTCollection.bytecode,
+    signer
+  )
+  nftCollection.deploy(
+    nftParams.nftName,
+    nftParams.nftSymbol,
+    nftParams.nftMaxSupply,
+    nftParams.nftMintCost,
+    nftParams.nftRevealBatchSize,
+    nftParams.nftRevealInterval,
+    VRF_COORDINATOR_V2_ADDRESS[chainId],
+    nftParams.vrfSubscriptionId,
+    VRF_GAS_LANE,
+    VRF_CALLBACK_GAS_LIMIT
+  )
+}
```

Comment:
This function needs to return at least the contract address or the whole contract because the create component needs to take the address from the deploy function and forward to the success page that will show the contract address.

Also will be good some error handling in case of unsuccessful deployment.